librz/bin/p/bin_sms.c

```diff
@@ -40,19 +41,24 @@ static bool check_buffer(RzBuffer *b) {
 }
 
 static bool load_buffer(RzBinFile *bf, RzBinObject *obj, RzBuffer *buf, Sdb *sdb) {
-	ut32 offset = 0;
-	if (!find_magic(bf->buf, &offset)) {
-		return false;
-	}
 	SMS_Header *sms_hdr = RZ_NEW0(SMS_Header);
 	if (!sms_hdr) {
 		return false;
 	}
-	rz_buf_read_at(bf->buf, offset, (ut8 *)sms_hdr, sizeof(SMS_Header));
+	if (!find_magic(bf->buf, &sms_hdr->offset)) {
+		free(sms_hdr);
+		return false;
+	}
+	sms_hdr->CheckSum = rz_read_le16(&sms_hdr->CheckSum);
```

Comment:
this cannot be done before you read the whole header via `rz_buf_read_at`, it must be done afterwards.